odl/operator/operator.py

```diff
@@ -95,57 +80,279 @@ def _default_apply(self, x, out, *args, **kwargs):
     -------
     `None`
     """
-    out.assign(self._call(x, *args, **kwargs))
+    out.assign(op._call_out_of_place(x, **kwargs))
 
 
-class _OperatorMeta(type):
+def _signature_from_spec(func):
+    """Return the signature of a python function as a string.
 
-    """Metaclass used by `Operator` to ensure correct methods.
+    Parameters
+    ----------
+    func : `function`
+        Function whose signature to compile
 
-    If either ``_apply()`` or ``_call()`` does not exist in the class
-    to be created, this metaclass attempts to add a default
-    implmentation.
-    This only works if the `Operator.range` is a
-    `LinearSpace`.
+    Returns
+    -------
+    sig : `str`
+        Signature of the function
     """
+    import sys
+    import inspect
 
-    def __new__(mcs, name, bases, attrs):
-        """Create a new instance."""
-        if '_call' in attrs and '_apply' in attrs:
-            pass
-        elif '_call' in attrs:
-            attrs['_apply'] = _default_apply
-        elif '_apply' in attrs:
-            attrs['_call'] = _default_call
+    # pylint: disable=deprecated-method,redefined-variable-type,no-member
+    py3 = (sys.version_info.major > 2)
```

Comment:
Another comment. Perhaps we should have a "odl_config.py" file somewhere that defines these things (they will likely be more)